### DISCUSSION TITLE
Getting started

### DISCUSSION THREAD

**Discussion by fsaris** [fsaris/EspHome-AwoX-BLE-mesh-hub#71] [General]
1. First get familiar with the basics of ESPHome https://esphome.io/guides/getting_started_hassio.html
  to ensure (when possible start with framework `esp-idf` then you can skip step 3.):
   - flashing/installing of a board/esp works
   - the board correctly connects to your wifi

2. Enable logging and validate you can see the logs when the device is running  (details about logging see https://esphome.io/components/logger)
   - Add `logger:` to your example yaml from step 1
   - install the updated yaml config 
   - From Esphome dashboard click on "Logs" ->  "wireless"
   - validate you see the debug logging. It should report info about WiFi connection 
 
3. Switch framework from `arduino` to `esp-idf` 
    - Update the `framework:` `type:` value in the yaml from `arduino` to `esp-idf`
    - install the updated yaml config __(this will require a wired connection due to partition table difference see [Bluetooth proxy documentation](https://esphome.io/components/bluetooth_proxy.html))__
    - validate it works by checking the logs again from previous step

4. Setup and test the mqtt connection
   - setup a mqtt broker 
   - remove the `api:` from your test yaml from step 1 by the mqtt config see https://esphome.io/components/mqtt.html

5.  Check if you can also see the logging now you switched to MQTT
    - validate the logs like you did in step 2
    - It should report info about MQTT and WiFi connection 

6. Now the build/install process and connections are working you can move on to adding the AwoX component 
    - bee sure to remove the following line from your yaml (to prevent memory issues)
       - `captive_portal:`
    - add the `external_components:` and all below from the example yaml to your yaml https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/blob/main/awox-ble-mesh-hub.yaml
    - install the updated yaml config
    - validate it works by checking the logs again from step 2
    - the logs should report all found devices etc
    - when devices are found and it could connect to one these will be reported to MQTT (and should show up in Home Assistant)

7. Next update the `awox_mesh:` `device_info:` in your yaml. https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/tree/main#devices

**Comment by echavet** [comment]
Hi and thanks for your sharing this component.
I don't understand the purpose of the `device_info:` section.

My understanding of your component is that it allows to make an esp32 become part of a AwoX mesh network, and to discover all AwoX registered devices so as to advertise them to ha MQTT. 

In this context, I don't see why there is a device_info section. 
Does it mean we have to first find the devices from the logs, within a first simple deployment and then add them manually in the yaml file to set them up with product id and options ?

I have another question:
- I'm using an esp32 s3 on which is already installed a component [to manage a mitsubishi heatpump through UART MitsubishiCN105ESPHome](https://github.com/echavet/MitsubishiCN105ESPHome)
- Will both MitsubishiCN105ESPHome and EspHome-AwoX-BLE-mesh-hub components be able to cohabit on the same esp32 ?

**Reply by fsaris** [reply]
> My understanding of your component is that it allows to make an esp32 become part of a AwoX mesh network, and to discover all AwoX registered devices so as to advertise them to ha MQTT.

It discovers all devices but the supported features are not known yet. At least I didn't find a method yet to get this info from the device itself.

https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub#device_info
`Unknown devices will be recognized as a dimmable device (DIM).`

> Does it mean we have to first find the devices from the logs, within a first simple deployment and then add them manually in the yaml file to set them up with product id and options ?

Yes more or less.

You do not need to check the logs. The product_id is visible in the device info/discovery message of each device. So if you are using HA you can find this by looking at the device info page of each device.

I would not recommend to combine the AwoX hub component. You could try it, but also depending on the number of devices that there are in your mesh it uses more memory. To handle and build the mqtt messages for instance.

**Reply by fsaris** [reply]
Will update the readme and getting started with more info about the product id

**Comment by echavet** [comment]
Thank you very much for your answer.

For information I've tried with an ESP32S3 to flash both components, my
Climate and your Mesh, and it works well.
I had to filter the MAC addresses of my 4 ATC Xiaomi Thermometers because
they were causing a nullPointer somewhere in a Ble lib.

I have few remarks:
- the entity names in homeassistant are based on the first discovered name
which results in something like this "unknown_device_type"
- You can have a look at my esphome mitsubishi integration as I saw you
want to make your sensors available through homeassistant API. You don't
have to, you can use the ESPHome integration which is the same at the end.
I did that for my climate component and for "extra components" named
stage_sensor, compressor_frequency_sensor...

But I'm surprised because you did it too for the main mesh component.
In my project, for example, adding a "stage_sensor:" inside the climate
component builds a new sensor available in the ESPHome integration.

Another way is to register your sensors with :
```
    this->iSee_sensor = new binary_sensor::BinarySensor();
    this->iSee_sensor->set_name("iSee sensor");
    this->iSee_sensor->publish_initial_state(false);
    App.register_binary_sensor(this->iSee_sensor);
```

https://github.com/echavet/MitsubishiCN105ESPHome/blob/main/components/cn105/climate.py

Thanks again


Le lun. 11 mars 2024 à 21:02, Frans Saris ***@***.***> a
écrit :

> Will update the readme and getting started with more info about the
> product id
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/discussions/71#discussioncomment-8751106>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYTHBDHAFOMA42UMHHHLG3YXYEW5AVCNFSM6AAAAABDOI3F4CVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DONJRGEYDM>
> .
> You are receiving this because you commented.Message ID:
> <fsaris/EspHome-AwoX-BLE-mesh-hub/repo-discussions/71/comments/8751106@
> github.com>
>


-- 
Eric

**Reply by fsaris** [reply]
Tnx for the feedback, will give it a look

**Reply by Kees48** [reply]
Hoi Frans, bedankt voor deze geweldige component! 🎉  
Ik heb alles draaiend met MQTT en zie mijn lampen via BLE, maar pairing lukt niet volledig.
Voorheen werkte het perfect maar na een herinstallatie van HA nietmeer.
- ESP ziet de lamp (`DISCOVERED` status), maar geen `active_connections`
- Geen `awox_mesh/.../discovery` topic op MQTT → Home Assistant vindt geen entiteiten
- Lamp draait via EGLO app, misschien blokkeert dat pairing?
- Zou het helpen om `api:` toe te voegen aan YAML zodat het apparaat verschijnt in Home Assistant via ESPHome?

Alvast dank voor je tijd en inzet! 🙌  
Groet, Kees

**Reply by fsaris** [reply]
Welke versie van esphome gebruik je? Er is een issue met de laatste versie #125 

Het gaat niet helpen om `api:` toe te voegen. Dat werkt niet samen met MQTT. En deze code werk momenteel niet via de api.

**Comment by Kees48** [comment]
Dank voor de snelle feed back. Ik werk met versie 2025.7.3. Kan ik ESPHome downgraden? Naar welke versie? Heeft het zin om de AWOX app af te sluiten en/of de lampen ed te resetten? Ik vind het trouwens jammer dat ik het energieverbruik van de Plug en PlugPlus niet kon uitlezen.

**Reply by fsaris** [reply]
Je kunt een oudere versie van esphome installeren (zie comments in #125).

De awox app sluiten is normaal gesproken niet nodig. En resetten van de lampen is niet nodig.

Ik heb zelf geen plug dus kan het uitlezen niet testen/debuggen. vandaar dat er (nog) geen support is voor het uitlezen van die waardes
